Docs lint/test traning data

m-vdb opened the pull request:
**Proposed changes**:
- Fix https://github.com/RasaHQ/rasa/issues/6399
- Tech writer API: use `yaml-rasa` or `yml-rasa` code blocks in docs
- Logic: created a simple python test that loads every docs file, parses it for codeblocks and validates the training data using `RasaYAMLReader`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

erohmensing commented:
bless you for working on this 🙏 i was looking into tagging all of them this morning -- happy to put in the manual work of doing that once this is ready

m-vdb commented:
@erohmensing it's ready! If you could flag all the training data that we want to validate in the docs, that'd be awesome. I've added a description to the PR so that it's easier to understand, but anyway it's not super hard to use. I've already added a CI step, and you can run the checks locally with `make test-docs`. It's highly probable that some use cases are not supported yet - bear with me!

m-vdb commented:
@erohmensing yes sure, will do 👍

erohmensing commented:
Otherwise I believe we are good! It can't seem to get some files for some reason, not sure what's up with that

m-vdb commented:
> It can't seem to get some files for some reason

@erohmensing which files?

erohmensing commented:
Oh wait, maybe someone updated them without updating them in the docs, looks like the autoconfig ones https://github.com/RasaHQ/rasa/runs/1014039899#step:13:113

m-vdb commented:
@erohmensing no it's a problem that has been introduced with this PR. Found the root cause, on it

erohmensing commented on .github/workflows/continous-integration.yml:
```diff
@@ -101,11 +101,14 @@ jobs:
-    - name: Test Docs 📃
+    - name: Build Docs 📃
       run: |
         poetry run python -c "from scripts import release; release.generate_changelog('major.minor.patch')"
```
wdyt about adding this directly to `make docs` so its easier to `make docs test-docs` locally?

m-vdb replied:
no we can't do that, it's used to generate the docs in the deployment pipeline

m-vdb replied:
but we could definitely modify test-docs to do that automatically

erohmensing replied:
ah makes sense 👍

erohmensing replied:
wait, doesn't the changelog still need to be generated though 🤔

m-vdb replied:
yes but it's not done like this. Look at the "major.minor.patch" argument :)

m-vdb replied:
I updated the workflow here: c55905a (and added a helper). It should be easier to use locally now 👍

erohmensing reviewed (approved):
🥳
